### PR TITLE
O(n) instead of O(n^2) string concactenation

### PR DESCRIPTION
**Pull request by ccuser44**
Ported my optimisation from FiOne [#19](https://github.com/Rerumu/FiOne/pull/19)

TLDR; concat function is always faster even with pcall (at least in Luau) https://github.com/Rerumu/FiOne/pull/19#issuecomment-2077599885

### CODE REVIEW

File: Source.lua [hunk 1]
```diff
@@ -1018,10 +1018,17 @@ local function luau_load(module, env, luau_settings)
 					local value = stack[inst.B]
 					stack[inst.A] = if value then value else inst.K or false
 				elseif op == 49 then --[[ CONCAT ]]
-					local s = ""
-					for i = inst.B, inst.C do
-						s ..= stack[i]
+					local B, C = inst.B, inst.C
+					local success, s = pcall(table.concat, stack, "", B, C)
```
Comment (TheGreatSageEqualToHeaven) [inline Source.lua:1022]: Can you define `table_concat` instead of `table.concat`

Comment (ccuser44) [reply Source.lua:1022]: Oh sure.

Comment (ccuser44) [reply Source.lua:1022]: Done.

**Review by TheGreatSageEqualToHeaven — APPROVED** [review]
LGTM

**Pull request merged**